stripe/public/javascript/app.js

```diff
@@ -0,0 +1,319 @@
+/**
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const STRIPE_PUBLISHABLE_KEY = '<YOUR STRIPE PUBLISHABLE KEY>';
+let currentUser = {};
+let customerData = {};
+
+/**
+ * Firebase auth configuration
+ */
+const firebaseUI = new firebaseui.auth.AuthUI(firebase.auth());
+const firebaseUiConfig = {
+  callbacks: {
+    signInSuccessWithAuthResult: function (authResult, redirectUrl) {
+      // User successfully signed in.
+      // Return type determines whether we continue the redirect automatically
+      // or whether we leave that to developer to handle.
+      return true;
+    },
+    uiShown: () => {
+      document.getElementById('loader').style.display = 'none';
+    },
+  },
+  signInFlow: 'popup',
+  signInSuccessUrl: '/',
+  signInOptions: [
+    firebase.auth.GoogleAuthProvider.PROVIDER_ID,
+    firebase.auth.EmailAuthProvider.PROVIDER_ID,
+  ],
+  credentialHelper: firebaseui.auth.CredentialHelper.NONE,
+  // Your terms of service url.
+  tosUrl: 'https://example.com/terms',
+  // Your privacy policy url.
+  privacyPolicyUrl: 'https://example.com/privacy',
+};
+firebase.auth().onAuthStateChanged((firebaseUser) => {
+  if (firebaseUser) {
+    currentUser = firebaseUser;
+    firebase
+      .firestore()
+      .collection('stripe_customers')
+      .doc(currentUser.uid)
+      .onSnapshot((snapshot) => {
+        if (snapshot.data()) {
+          customerData = snapshot.data();
+          startDataListeners();
+          document.getElementById('loader').style.display = 'none';
+          document.getElementById('content').style.display = 'block';
+        }
```

Comment:
While we're here, let's fix a small bug. We have an `if (snapshot.data())`, but can you please add an `else` that console.logs or similar to indicate that no customer record for this user ID was found? I ran into this because I signed in with a user that had already been created on my project, so the auth-triggered function never ran.